.github/workflows/cherryPick.yml

```diff
@@ -173,6 +173,27 @@ jobs:
             echo "HAS_CONFLICTS=true" >> "$GITHUB_OUTPUT"
           fi
 
+      # This avoids a race condition where a staging deployment (potentially with a higher version)
+      # could complete before an ongoing production deployment (with a lower version),
+      # leading to failures when the production version is submitted to app stores.
+      - name: Check for active production deployments before pushing to staging
+        if: ${{ inputs.TARGET == 'staging' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          DEPLOY_WORKFLOW_FILENAME="deploy.yml"
+          ACTIVE_PRODUCTION_RUNS=$(gh api repos/${{ github.repository }}/actions/workflows/${DEPLOY_WORKFLOW_FILENAME}/runs \
+            --jq --arg branch "production" \
+            '.workflow_runs[] | select(.head_branch == $branch and (.status == "in_progress" or .status == "queued")) | .id')
```

Comment:
We have a fail check `shellcheck reported issue in this script: SC2016:info:4:3: Expressions don't expand in single quotes, use double quotes for that`.

Maybe we should use double quotes and escaped `$` as `\$branch` and inner double quotes, e.g., `\"in_progress\"`.